Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -305,31 +305,63 @@ public async Task<string> WalkTree(string treeNodeKey)
                         this.walkTreeCts.Token.ThrowIfCancellationRequested();
                     }
 
-                    await this.Parameters.Callbacks.BeforeVisitNode(
-                        this.Parameters.SessionId,
-                        current,
-                        await this.EvaluateDynamicProperty(this.Schema.Tree[current].Properties, null),
-                        this.Parameters.UserContext,
-                        this.Parameters.TreeName,
-                        this.Parameters.RootSessionId,
-                        this.walkTreeCts.Token).ConfigureAwait(false);
+                    TreeNodeContext treeNodeContext = null;
+                    if (this.Parameters.CallbacksV2 != null)
+                    {
+                        // If applicable, use the new CallbacksV2 with ITreeWalkerCallbacksV2
+                        // Evaluate the dynamic properties that are used by the actionTask.
+                        treeNodeContext = new TreeNodeContext(
+                            this.Parameters.SessionId,
+                            treeNodeKey,
```

Comment:
>treeNodeKey [](start = 28, length = 11)

current instead of treeNodeKey

Reply:
wow, this is a good bug! thanks

---
In reply to: [504162909](https://github.com/microsoft/Forge/pull/32#discussion_r504162909) [](ancestors = 504162909)